Content.Server/Guardian/GuardianSystem.cs

```diff
@@ -0,0 +1,183 @@
+               args.Message.AddMarkup("\n" + $"[color=#ba1919]{usedstring}[/color]");
+           }
+        }
+
+        /// <summary>
+        /// Adds the guardian host component to the user and spawns the guardian inside said component
+        /// </summary>
+        private void OnGuardianCreated(EntityUid uid, GuardianCreatorComponent component, UseInHandEvent args)
+        {
+            //Only works if the guardian isn't already present. NOTE: this is up to preference, we can probably later allow
+            //a person to host multiple guardians, however the idea is that in a normal setting you won't get the opportunity to
+            if (!args.User.TryGetComponent<GuardianHostComponent>(out var guardianComponent))
+            {
+                if (args.User.TryGetComponent<SharedActionsComponent>(out SharedActionsComponent? action))
+                {
+                    if (component.Used == false)
+                    {
+                        var hostcomp = args.User.EnsureComponent<GuardianHostComponent>();
+                        var guardian = EntityManager.SpawnEntity(component.GuardianType, hostcomp.Owner.Transform.Coordinates);
+                        hostcomp.GuardianContainer.Insert(guardian);
+                        hostcomp.Hostedguardian = guardian.Uid;
+                        //Takes the guardian component on the supposed guardian entity and fills out it's synced parametres
+                        if (guardian.TryGetComponent<GuardianComponent>(out GuardianComponent? guardiancomp))
+                        {
+                            guardiancomp.Host = args.User.Uid;
+                            //Grant the user the recall action and notify them
+                            action.Grant(ActionType.ManifestGuardian);
+                            SoundSystem.Play(Filter.Entities(uid), "/Audio/Effects/guardian_inject.ogg", uid);
+                            _popupSystem.PopupCoordinates(Loc.GetString("guardian-created"), hostcomp.Owner.Transform.Coordinates, Filter.Entities(args.User.Uid));
+                            //Exhaust the activator
+                            component.Used = true;
+                        }                   
+                    }
+                    else
+                    {
+                        _popupSystem.PopupCoordinates(Loc.GetString("guardian-activator-empty-invalid-creaton"), args.User.Transform.Coordinates, Filter.Entities(args.User.Uid));
+                    }
+                }
+                else
+                {
+                    _popupSystem.PopupCoordinates(Loc.GetString("guardian-no-actions-invalid-creation"), args.User.Transform.Coordinates, Filter.Entities(args.User.Uid));
+                }
+            }
+            else
+            {
+                _popupSystem.PopupCoordinates(Loc.GetString("guardian-already-present-invalid-creation"), args.User.Transform.Coordinates, Filter.Entities(args.User.Uid));
+            }
+        }
```

Comment:
As above. This also has a bunch of code that should be on GuardianHostComponent initialisation and not this so it can be VV'd properly.

Reply:
Not sure how to properly code this one on the initialization. Specifically, how am I gonna pass the data from the system into the it?

Reply:
I ended up just cleaning it up for now (it's being initialised on the system so passing the data as args is a non-issue).